src/Avalonia.Visuals/Rendering/RenderLoop.cs

```diff
@@ -91,7 +91,19 @@ private void TimerTick(TimeSpan time)
             {
                 try
                 {
-                    if (_items.Any(item => item.NeedsUpdate) &&
+                    bool needsUpdate = false;
```

Comment:
It might be worth to start implementing specialized versions of LINQ methods.

E. g. `bool ListAny<T>(this List<T> list, Func<T, bool> cb)` shouldn't allocate.

Reply:
Well, in this case we can avoid allocations since we can cache `Func<T,bool>` but this won't work universally as user still needs to allocate new delegate and closure.

Reply:
I'm pretty sure `item => item.NeedsUpdate` is cached in a static field, e. g. https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4oDchGAzOTkuVqQMKkDehpv5DFZhhSkAsgAoAMgEsAzgBcAPNIB28gHykANgoCUpALyad8gHQyFAQRUBPcQiMJ1AVl00CfUjz71GzYBAQWqQW8tY2igAq6uLyABZyIXJK0doKADTkzlGZAUGaYMC63rzcHp58AGbQMACGYHHiAG61UKQIpKpp8sXlFRXSleKF9rq9/RM+AOyk8lAArjDukxgzlbU6SyWkAL7bhHsEhEA

Reply:
What I mean is that in such simple cases it will work, but as soon as that delegate requires state it won't be cached. So this will not help in most of the cases.

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4oDchGAzOTkuVqQMKkDehpv5DFZhhSkAsgAoAMgEsAzgBcAPNIB28gHykANgoCUPPtwJ8T2hQDoZCgIIqAnuIQBedQnUBWXTWOneB3wAKAIZQQQC2MPJQ0gBeMAAm4jryADSknt4mAL7+pLn0jEIiwaERUbEJUnJKqhpmqaS1pABuQVr6PrxGvrzJltW2DgikLqTDmq3tmXw5nXlzBYKkwBAQWqRW8oOKACrq4vIAFnIb1buayWkY7rtpK2uaYMAdJt09AGbQMEFgh+KtUDGjRU9RePRM0ne4iejl0YPBvgwAHZSFEAK4waaIlHvNqyTG5WYmQizQhAA